test/integration/gateway/gateway.go

```diff
@@ -0,0 +1,131 @@
+/*
```

Comment:
it may make sense to move this folder under test/integration/pkg to be consistent with the other pkg integration tests.

Reply:
ok, I'll move this